internal/server/server.go

```diff
@@ -1,2 +1,179 @@
-// TODO: document the package.
+// This package containts server implementation to receive events in format described in
+// thermomatic/internal/client/reading.go.
 package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/spin-org/thermomatic/internal/client"
+	"github.com/spin-org/thermomatic/internal/imei"
+)
+
+const (
+	loginTimeout = 1 * time.Second
+	readTimeout = 2 * time.Second
+)
+
+var errReadTimeout = errors.New("read timeout")
+
+type Server struct {
+	create chan net.Conn
```

Comment:
Why do we need these channels?

Reply:
To track open connections in internal map

Reply:
This just slows everything down; perhaps a `stats` top level package that keeps this updated via atomic means?